### PR TITLE
Dw 2427 decryption

### PR DESCRIPTION
**Pull request by DanielChicot**
Data is decrypted and decompression processor merged in.

### CODE REVIEW

File: docker/s3-init/backup.jsonl [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{ "name": "Alice", "age": 21 }
```
Comment (UdayChokkam) [inline docker/s3-init/backup.jsonl]: Can we get rid of this file ?

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -2,4 +2,4 @@ distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https://nexus3.ucds.io/repository/thirdparty/org/gradle/distributions/gradle/5.5.1/gradle-5.5.1-bin.zip
+distributionUrl=https://downloads.gradle.org/distributions/gradle-5.5.1-bin.zip
```
Comment (UdayChokkam) [inline gradle/wrapper/gradle-wrapper.properties]: Can we change this to uc expected nexus ?

File: src/test/kotlin/app/batch/EncryptionMetadataProcessorTest.kt [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+import app.exceptions.MetadataException
+import org.junit.Test
+
+import org.junit.Assert.*
+import java.io.ByteArrayInputStream
+
+class EncryptionMetadataProcessorTest {
+
+    @Test
+    fun processOk() {
```
Comment (UdayChokkam) [inline src/test/kotlin/app/batch/EncryptionMetadataProcessorTest.kt]: Can the test names be a little more descriptive?

**Review by UdayChokkam — APPROVED** [review]

File: src/test/kotlin/app/services/impl/HttpKeyServiceTest.kt [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+    @Before
+    fun init() {
+        this.keyService.clearCache()
+        reset(this.httpClientProvider)
+    }
+
+    @Test
+    fun testDecryptKeyOk() {
+        val responseBody = """
```
Comment (mark-r-m) [inline src/test/kotlin/app/services/impl/HttpKeyServiceTest.kt:47]: Can we please have test names like "decryptKey_will_call_dks_and_return_data_when_input_is_valid" or something simmilar?

**Review by mark-r-m — CHANGES_REQUESTED** [review]

File: src/test/kotlin/app/services/impl/HttpKeyServiceTest.kt (hunk 4 above)
Comment (mark-r-m) [reply src/test/kotlin/app/services/impl/HttpKeyServiceTest.kt:47]: As a sample see what we now have in HTME

**Pull request merged**